repair_shards.c

```diff
@@ -209,3 +204,67 @@ RecreateTableDDLCommandList(Oid relationId, int64 shardId)
 
 	return extendedRecreateCommandList;
 }
+
+
+/*
+ * CopyDataFromFinalizedPlacement connects to an unhealthy placement and
+ * directs it to copy the specified shard from a certain healthy placement.
+ * This function assumes that the unhealthy placement already has a schema
+ * in place to receive rows from the healthy placement. This function returns
+ * a boolean indicating success or failure.
+ */
+static bool
+CopyDataFromFinalizedPlacement(ShardPlacement *placementToRepair,
```

Comment:
1. Fun fact: my unit tests pass even if I switch the order of arguments here. Side effect of hacking together a shard repair test using a single PostgreSQL instance. Chalk it up as another vote for "need to make better integration testing rig, eventually". Good catch, that was a bug that manual testing would have caught (which I did a few months ago but had refactored since, trusting my test).
2. Yes, changing.